src/generic/armcm_boot.c

```diff
@@ -27,12 +28,39 @@ extern uint32_t _stack_end;
 void
 ResetHandler(void)
 {
+    __disable_irq();
+
+    // Explicitly load the stack pointer (for some bootloaders that don't)
+    asm volatile("mov sp, %0\n bx %1"
+                 : : "r"(&_stack_end), "r"(ResetHandlerC));
+}
+
+void
+__noreturn
+ResetHandlerC(void)
+{
+    // Clear all enabled user interrupts and user pending interrupts
+    for (uint8_t i=0; i < sizeof(NVIC->ICER)/sizeof(NVIC->ICER[0]); i++) {
```

Comment:
I'd avoid uint8_t on arm processors (they cause gcc to emit poor code).  I'd just use `int`.  I'd also put the `int i` declaration outside the for loop as some gcc variants complain about that C99 variant in some cases.  Use `ARRAY_SIZE()` helper instead of manually calling `sizeof()`.